### PR TITLE
Add `from` to `GetTransaction` response

### PR DESCRIPTION
**Pull request by zeroqn**

### CODE REVIEW

File: jsonrpc-types/Cargo.toml [hunk 1]
```diff
@@ -16,6 +17,9 @@ serde_derive = "1.0"
 rustc-serialize = "0.3"
 uuid = { version = "0.7", features = ["v4"] }
 
+[dev-dependencies]
+cita-crypto = { path = "../cita-crypto" }
```
Comment (chaoticlonghair) [inline jsonrpc-types/Cargo.toml]: The [`cita-crypto` features] should be added to [`jsonrpc-types` features].

[`cita-crypto` features]: https://github.com/cryptape/cita-common/blob/4b5022b/cita-crypto/Cargo.toml#L12-L19
[`jsonrpc-types` features]: https://github.com/cryptape/cita-common/blob/32871ee/jsonrpc-types/Cargo.toml#L19-L26

File: jsonrpc-types/Cargo.toml [hunk 2]
```diff
@@ -6,6 +6,7 @@ authors = ["Cryptape Technologies <contact@cryptape.com>"]
 [dependencies]
 jsonrpc-types-internals = { path = "internals" }
 cita-types = { path = "../cita-types" }
+util = { path = "../util" }
```
Comment (chaoticlonghair) [inline jsonrpc-types/Cargo.toml]: The [`util` features] should be added to [`jsonrpc-types` features].

[`util` features]: https://github.com/cryptape/cita-common/blob/4dc0912/util/Cargo.toml#L37-L41
[`jsonrpc-types` features]: https://github.com/cryptape/cita-common/blob/32871ee/jsonrpc-types/Cargo.toml#L19-L26

**Review by chaoticlonghair — COMMENTED** [review]
I would not approve this PR, since you add the `util` into dependencies.

If your draw a map of dependencies for all crates in this repo, you will find a very complex topological graph.
I think we should reduce the complexity of dependencies, too avoid some problems, such as the diamond dependency.

`jsonrpc-types` is just an interface, it is in low-level.
`util` has a lot of advanced features, it is in top-level.
Do **NOT** let `jsonrpc-types` to be dependent on `util`.

If a user write a JSON-RPC client, he should **NOT** have to compile `rocksdb` (in `util`, now).

@u2 your opinion?

File: jsonrpc-types/Cargo.toml (hunk 1 above)
Comment (u2) [reply jsonrpc-types/Cargo.toml]: We have talked about the solution about this, there are two huge tasks including:
1. Split `util` crate to some small crates.
2. The types of different services should be in independent types. And If there are type conversions between different services, put the type conversions in another crate, or put it in the right place where it just used.

Comment (u2) [reply jsonrpc-types/Cargo.toml]: We can move this conversion to proto firstly and refactor it later.

**Comment by zeroqn** [comment]
All done.

**Comment by u2** [comment]
LGTM

**Review by u2 — DISMISSED** [review]

**Comment by zeroqn** [comment]
All pass. It should be mergeable.

File: jsonrpc-types/Cargo.toml [hunk 3]
```diff
@@ -15,12 +15,13 @@ serde_json = "1.0"
 serde_derive = "1.0"
 rustc-serialize = "0.3"
 uuid = { version = "0.7", features = ["v4"] }
+cita-crypto = { path = "../cita-crypto", optional = true }
```
Comment (chaoticlonghair) [inline jsonrpc-types/Cargo.toml:18]: Not perfect, the `optional` is useless.

**Review by chaoticlonghair — APPROVED** [review]

File: jsonrpc-types/Cargo.toml (hunk 3 above)
Comment (zeroqn) [reply jsonrpc-types/Cargo.toml:18]: Yea. But there's no way around it. Let's refactor code later.

Comment (chaoticlonghair) [reply jsonrpc-types/Cargo.toml:18]: I mean, if you remove this `optional = true`, this PR still can be passed.
At least, one feature have to be selected, so this crate is not `optional`, it is **necessary**.

**Pull request merged**